homework/assignment_1/assignment_1.py

```diff
@@ -0,0 +1,233 @@
+print("Quartiles of sepal length", iris["sepal_length"].quantile([0.25, 0.5, 0.75]))
+print("\n")
+
+# Sepal width
+print("Sepal width stats:")
+print("Mean of sepal width: ", iris["sepal_width"].mean())
+print("Max of sepal width:", iris["sepal_width"].max())
+print("Min of sepal width:", iris["sepal_width"].min())
+print("Quartiles of sepal width", iris["sepal_width"].quantile([0.25, 0.5, 0.75]))
+print("\n")
+
+# Pedal length
+print("Pedal length stats:")
+print("Mean of pedal length: ", iris["pedal_length"].mean())
+print("Max of pedal length:", iris["pedal_length"].max())
+print("Min of pedal length:", iris["pedal_length"].min())
+print("Quartiles of pedal length", iris["pedal_length"].quantile([0.25, 0.5, 0.75]))
+print("\n")
+
+# Pedal width
+print("Pedal width stats:")
+print("Mean of pedal width: ", iris["pedal_width"].mean())
+print("Max of pedal width:", iris["pedal_width"].max())
+print("Min of pedal width:", iris["pedal_width"].min())
+print("Quartiles of pedal width", iris["pedal_width"].quantile([0.25, 0.5, 0.75]))
+print("\n")
+
+print(np.mean(iris["sepal_length"]))
+
+
+# plotting different classes against each other
+# plot 1 - scatter plot
+df = px.data.iris()  # iris is a pandas DataFrame
+scatter = px.scatter(
+    df,
+    x="sepal_width",
+    y="sepal_length",
+    color="species",
+    size="petal_length",
+    hover_data=["petal_width"],
+    title="Scatter plot of iris data set",
+)
+scatter.write_html(file="scatter_plot.html", include_plotlyjs="cdn")
+# scatter.show()
+
+# plot 2 - violin plot
+df = px.data.iris()
+violin = px.violin(
+    df,
+    y="sepal_width",
+    color="species",
+    violinmode="overlay",
+    title="Violin plot of iris data set",  # draw violins on top of each other
+    # default violinmode is 'group' as in example above
+    hover_data=df.columns,
+)
+violin.write_html(file="violin_plot.html", include_plotlyjs="cdn")
+# violin.show()
+
+# plot 3 - scatter matrix
+df = px.data.iris()
+scatter_matrix = px.scatter_matrix(
+    df,
+    dimensions=["sepal_width", "sepal_length", "petal_width", "petal_length"],
+    color="species",
+    symbol="species",
+    title="Scatter matrix of iris data set",
+    labels={col: col.replace("_", " ") for col in df.columns},
+)  # remove underscore
+scatter_matrix.update_traces(diagonal_visible=False)
+scatter_matrix.write_html(file="scatter_matrix.html", include_plotlyjs="cdn")
+# scatter_matrix.show()
+
+# plot 4 - distribution plot
+df = px.data.iris()
+hist = px.histogram(
+    df,
+    x="sepal_width",
+    y="sepal_length",
+    color="species",
+    marginal="violin",
+    hover_data=df.columns,
+    title="Distribution plot of iris data set",
+)
+hist.write_html(file="hist_plot.html", include_plotlyjs="cdn")
+# hist.show()
+
+# plot 5 - boxplot
+df = px.data.iris()
+box = px.box(df, y="sepal_length", color="species", title="Boxplot of iris data set")
+box.write_html(file="box_plot.html", include_plotlyjs="cdn")
+# box.show()
+
```

Comment:
Might want to uncomment box.show() for ease of use, or look into different plotly renders to simplify displays

Reply:
uncommented